baseplate/secrets/fetcher.py

```diff
@@ -0,0 +1,282 @@
+"""Daemon that fetches secrets from Vault and writes them to disk.
+
+This daemon looks for configuration in an INI file passed on the command line.
+The file should contain a section looking like:
+
+    [secret-fetcher]
+    vault.url = https://vault.example.com:8200/
+    vault.role = my-server-role
+
+    output.path = /var/local/secrets.json
+    output.owner = www-data
+    output.group = www-data
+    output.mode = 0400
+
+    secrets =
+        secret/one,
+        secret/two,
+        secret/three,
+
+where each secret is a path to look up in Vault. The daemon authenticates with
+Vault as the EC2 server it is running on. Vault can map that context to
+appropriate policies accordingly.
+
+The secrets will be read from Vault and written to output.path as a JSON file
+with the following structure:
+
+    {
+        "expiration": "2017-05-05T09:42.143:35Z",
+        "secrets": {
+            "secret/one": {...},
+            "secret/two": {...},
+            "secret/three": {...}
+        },
+        "vault_token": "9da4241c-3460-11e7-84ac-0e9f9d32522f"
+    }
+
+The vault_token can be used for direct communication with Vault using the
```

Comment:
Is the vault_token going to be per-instance, app, or role? In other words, can I piggy back off of this token to store cubbyhole data that is accessible across the app?